### DISCUSSION TITLE
pytest: using Screen to select an element from a dropdown?

### DISCUSSION THREAD

**Discussion by cristianmtr** [zauberzeug/nicegui#2982] [Q&A]
### Question

Hey,

Is there an API for selecting from a dropdown?

I can only find `.click` and some others, but can't figure out how to piece them together to mock an interaction with a dropdown (`ui.select`).

Looking at the code [here](https://github.com/zauberzeug/nicegui/blob/main/nicegui/testing/screen.py#L24)

**Comment by falkoschindler** [comment]
Hi @cristianmtr,

You can simply click on the current value and then on one of the options.
See https://github.com/zauberzeug/nicegui/blob/main/tests/test_select.py

**Comment by cristianmtr** [comment]
It doesn't seem to work for me. Is there any way to pass an id to an element that I can then use in the test to get the specific element?

`element.id = "custom_id"` causes a crash further on

**Reply by falkoschindler** [reply]
You can use `screen.find_element(element: ui.element)` to find an element in the DOM if you have access to the NiceGUI element.
Or you add a custom class and use `screen.find_by_class`.
